corehq/motech/repeaters/tests/test_proxy_models.py

```diff
@@ -183,3 +219,212 @@ def test_repeaters_are_synced_to_couch(self):
         )
+
+
+class ModelAttrEqualityHelper(SimpleTestCase):
+    class DummySQLModel(models.Model, SyncSQLToCouchMixin):
+        pass
+
+    class DummyCouchModel(Document, SyncCouchToSQLMixin):
+        pass
+
+    @classmethod
+    def _get_user_defined_attrs(cls, model_cls, dummy_model):
+        model_attrs = dir(dummy_model)
+        return {item[0]
+                for item in inspect.getmembers(model_cls)
+                if item[0] not in model_attrs}
+
+    @classmethod
+    def get_sql_attrs(cls, model_cls):
+        return cls._get_user_defined_attrs(model_cls, cls.DummySQLModel)
+
+    @classmethod
+    def get_cleaned_couch_attrs(cls, couch_model_cls):
+        couch_attrs = cls._get_user_defined_attrs(couch_model_cls, cls.DummyCouchModel)
+        extra_attrs = cls._couch_only_attrs()
+        new_attrs = cls._sql_only_attrs()
+        return (couch_attrs - extra_attrs).union(new_attrs)
+
+    @classmethod
+    def _couch_only_attrs(cls):
+        return set()
+
+    @classmethod
+    def _sql_only_attrs(cls):
+        return set()
+
+
+class TestRepeaterModelsAttrEquality(ModelAttrEqualityHelper):
+
+    def test_have_same_attrs(self):
+        couch_attrs = self.get_cleaned_couch_attrs(Repeater)
+        sql_attrs = self.get_sql_attrs(SQLRepeater)
+        self.assertSetEqual(couch_attrs - sql_attrs, set())
+        self.assertSetEqual(sql_attrs - couch_attrs, set())
+
+    @classmethod
+    def _couch_only_attrs(cls):
+        return {
+            # removed
+            'last_success_at',
+            'sql_repeater',
+            'failure_streak',
+            'started_at',
+            # renamed
+            'paused',
+            # connection setting props
+            'plaintext_password', 'username', 'notify_addresses_str', 'create_connection_settings', 'name', 'url',
+            'verify', 'skip_cert_verify', 'password', 'auth_type',
+            # not required in sql
+            'by_domain', 'base_doc',
+            'get_class_from_doc_type', 'started_at', '_get_connection_settings',
+            'clear_caches',  # will see if we need it as per requirement
+            # not used
+            'get_attempt_info'
+        }
+
+    @classmethod
+    def _sql_only_attrs(cls):
+        return {
+            'repeater_id', 'set_next_attempt', 'next_attempt_at',
+            'is_ready', 'options', '_repeater_type', 'last_attempt_at', 'repeat_records_ready', 'repeat_records',
+            'all_objects', 'reset_next_attempt', 'is_deleted', 'PROXY_FIELD_NAME', 'Meta', 'repeater',
+            'get_request_method_display'  # added by django choicefield in models
+        }
+
+
+class TestCaseRepeaterAttrEquality(TestRepeaterModelsAttrEquality):
+    def test_have_same_attrs(self):
+        couch_attrs = self.get_cleaned_couch_attrs(CaseRepeater)
+        sql_attrs = self.get_sql_attrs(SQLCaseRepeater)
+        self.assertSetEqual(couch_attrs - sql_attrs, set())
+        self.assertSetEqual(sql_attrs - couch_attrs, set())
```

Comment:
Could this be simplified to a single assertion using set [`symmetric_difference`](https://docs.python.org/3/library/stdtypes.html#frozenset.symmetric_difference)?
```suggestion
        self.assertEqual(couch_attrs ^ sql_attrs, set())
```
Also, can use `assertEqual` since that "method is used by default when comparing sets or frozensets with [assertEqual()](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual)." ([source](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertSetEqual))

The same applies to other tests below.

Reply:
I did not know about symmetric_difference's existence before. Thanks for sharing it 😄 
I have implemented it https://github.com/dimagi/commcare-hq/pull/31772/commits/3489608585239a6cc4db921a0c53f2b5dc55170c